### ISSUE TITLE
Got stuck when running train shapes demo with Multi-GPU

### ISSUE THREAD

**Issue by Chenfengldw** [matterport/Mask_RCNN#1416]
In the train shape demo, I set GPU_COUNT = 4 in the class ShapesConfig, and run the code on my 4-GPU machine. According to previous issues, I also set my Keras version to 2.1.3

After a while, the code was stuck at
`I tensorflow/stream_executor/dso_loader.cc:152] successfully opened CUDA library libcublas.so.10.0 locally`  and it didn't give any output.

Meanwhile, the process occupies 4 GPUs with almost full memory usage but with 0% GPU compute.

**Comment by Icecream98** [comment]
I have the same issue, have you solved it yet?

**Comment by Chenfengldw** [comment]
> I have the same issue, have you solved it yet?

Not yet, maybe some bugs in multi GPU mode.

**Comment by Icecream98** [comment]
> > I have the same issue, have you solved it yet?
> 
> Not yet, maybe some bugs in multi GPU mode.

yeah, I found my training images too big for the network(3M~4M per image and ~400 target per image), so I cut them into pieces and then solve this problem.

**Comment by DecentMakeover** [comment]
@Icecream98 hey i am facing a similar issue but while testing, when i run the demo,ipynb , when i infer using 2 GPU's it works fine , but when i scale that to 4GPUS i get

InvalidArgumentError: Input to reshape is a tensor with 600 values, but the requested shape has 4800 [[{{node tower_0_4/mask_rcnn/mrcnn_detection/Reshape_1}} = Reshape[T=DT_FLOAT, Tshape=DT_INT32, _device="/job:localhost/replica:0/task:0/device:GPU:0"](tower_0_4/mask_rcnn/mrcnn_detection/packed, tower_0_4/mask_rcnn/mrcnn_detection/Reshape_1/shape)]] 


Any ideas?
Thanks
